### ISSUE TITLE
Custom icon only works on ui-gmap-markers and not a singular marker

### ISSUE THREAD

**Issue by intellix** [angular-ui/angular-google-maps#1408]
Trying to use a custom icon, but can only get it working when using `<ui-gmap-markers>`. I can see it working on your example, which is confusing as I don't see the difference/cause.

I've created a simple plnkr with both, which highlights the issue: http://plnkr.co/edit/6tKWteGE6iu0pEY9d21F?p=preview

**Comment by leocaseiro** [comment]
+1

**Comment by tricmir** [comment]
+1

**Comment by leocaseiro** [comment]
If you set the icon inside options, it works for ui-gmap-marker: http://plnkr.co/edit/2JxtGRoc3jcbf4hM0AA8?p=preview

**Comment by solidspark** [comment]
I can probably look at this, it's a simple model issue.

**Comment by nmccready** [comment]
@solidspark thanks

**Comment by ecbrodie** [comment]
+1, I am also seeing the icon attribute not working on the marker directive. I have to use options for it. A fix to bring the implementation back to matching the documentation is much appreciated.

**Comment by krewmarco** [comment]
wow, I wasted a lot of time on this

**Comment by nmccready** [comment]
Why not just use markers?

**Comment by leocaseiro** [comment]
marker = 1
marker = 2+

Anyway, as I said before, set the icon inside options. If you just update the documentation, so that would be ok.

**Comment by krewmarco** [comment]
Yes, markers work great for this.  I just couldn't find it in the docs.

**Comment by nmccready** [comment]
I am sure this fix is dead simple; is anyone else still looking into the code base for this?

**Comment by nmccready** [comment]
Problem code is either in here https://github.com/angular-ui/angular-google-maps/blob/master/src/coffee/directives/api/models/child/marker-child-model.coffee 

or here

https://github.com/angular-ui/angular-google-maps/blob/master/src/coffee/directives/api/marker.coffee

**Comment by adyba** [comment]
I don't really care whether custom icon path is defined in `marker.icon` or `marker.option.icon`. 
It looks so trivial that this undocumented "feature" results in hours of ![](http://fc07.deviantart.net/fs14/f/2007/105/3/f/Head_Smash_Avatar_by_Masterswordman.gif)

**Comment by nmccready** [comment]
lol, I am putting the lines out there so some one else can work on this besides me. So keep banging your head while you wait or attempt to solve the issue. I am just trying to help while many complain.

**Comment by nmccready** [comment]
While your at it document the problem on the wiki, and or website which you can contribute to.

**Comment by leocaseiro** [comment]
@nmccready I've tried to send a pull request to the gh-pages, but honestly, I'm not sure how to do it. You can check my [commit](https://github.com/leocaseiro/angular-google-maps/commit/abb3bdd45cbf815cc7b5f18c4a0cd1ede1c029dc) if you're able to update the documentation.

**Comment by leocaseiro** [comment]
perhaps this issue can be closed

**Comment by nmccready** [comment]
I can make it work with singular, just got to do it. It's not hard , just gotta find the time.

**Comment by nmccready** [comment]
So leave it open as an enhancement.

**Comment by leocaseiro** [comment]
:+1:

**Comment by andreibenhur** [comment]
Any ideia when we will have a new release with this enhancement?

**Comment by sameershenai** [comment]
Check your closing element its markers not marker :)
     ```<ui-gmap-marker idKey="marker.id"
                      coords="marker.coords"
                      icon="marker.icon"></ui-gmap-markers>```
